### PR TITLE
Login using CH Connect

### LINKED ISSUE #5: Spring Security
Login using [WISVCH/connect](https://github.com/WISVCH/connect).

### PR DESCRIPTION
**Pull request by praseodym**
Resolves #5

- [x] Documentation for configuration
- [x] Use `JsonFileRegisteredClientService` to save dynamic client registration
- [x] Ensure that test builds do not register a new client
- [x] Add proper authorisation (do not force log in for all requests)
- [x] ~~Signed/encrypted requests for production~~

### CODE REVIEW

**Comment by praseodym** [comment]
- [ ] Document configuration
- [ ] Use `JsonFileRegisteredClientService` to save dynamic client registration
- [ ] Ensure that test builds do not register a new client

File: src/main/resources/templates/index.html [hunk 1]
```diff
@@ -33,7 +33,7 @@
         <h1>Events</h1>
 
         <p>
-            Welcome to the CH event registration app!
+            Welcome to the CH event registration app, <span th:text="${userInfo.givenName}">Derp</span>!
```
Comment (jgadelange) [inline src/main/resources/templates/index.html]: Is this only visible for logged in users?

Comment (praseodym) [inline src/main/resources/templates/index.html]: Currently everyone has to log on. We should probably change that :)

Comment (RickWieman) [inline src/main/resources/templates/index.html]: Especially if we want guests to register :wink:

**Comment by jgadelange** [comment]
image

**Comment by praseodym** [comment]
When do you see this error?

**Comment by jgadelange** [comment]
After I return from connect.ch.tudelft.nl

**Comment by jgadelange** [comment]
Ah, I used 127.0.0.1 instead of localhost :wink:

**Comment by praseodym** [comment]
Hmm still interesting that this would give an issuer whitelist error though.

**Comment by RickWieman** [comment]
Cool, tested it and it works nice! :sunglasses:

**Pull request merged**